src/vm/database/sqlite.rs

```diff
@@ -352,29 +364,59 @@ impl <'a> ContractDatabase <'a> {
             key_type: TypeSignature::deserialize(&key_type) })
     }
 
-    pub fn create_token(&mut self, contract_name: &str, token_name: &str) {
-        self.execute("INSERT INTO tokens_table (contract_name, token_name) VALUES (?, ?)",
-                     &[contract_name, token_name]);
+    pub fn create_token(&mut self, contract_name: &str, token_name: &str, total_supply: &Option<i128>) {
+        let contract_name_owned = contract_name.to_string();
+        let contract_name_owned = token_name.to_string();
+        
+        let params: [&ToSql; 3] = [&contract_name, &token_name, total_supply];
+        self.execute("INSERT INTO tokens_table (contract_name, token_name, total_supply) VALUES (?, ?, ?)",
+                     &params);
+        if total_supply.is_some() {
```

Comment:
Shouldn't this be `is_none()`?  It doesn't make sense to me that you're inserting a value of `0 as i128` as the token supply?

Reply:
No -- I am probably naming these variables wrong, but that 0 is the "current supply" (circulating supply?)

Reply:
So, total supply is a property of the token (hence, stored directly in the `tokens_table`), while current supply is updated on `ft-mint!` calls (stored in the `data_table`).

Reply:
Ah, okay, got it :+1: